### DISCUSSION TITLE
Switching between HEAD and released version

### DISCUSSION THREAD

**Discussion by meling** [Homebrew/discussions#838] [Everyday usage]
Hi. First off. Thanks for homebrew. I much appreciate this community's contributions.

I wonder what is the right way to switch between a HEAD and a released version. For my specific example, I want to have the latest release version of `go` installed, while also occasionally switching to and playing around with the `HEAD` version.

I've installed both, and that's fine.
```shell
% ls /usr/local/Cellar/go
1.15.8
HEAD-2f0da6d
```
 But after unlinking the released version, I'm unable to link back to the `HEAD` version:
```shell
% brew unlink go
Unlinking /usr/local/Cellar/go/1.15.8... 3 symlinks removed.
% brew install go --HEAD
Warning: go HEAD-2f0da6d is already installed, it's just not linked.
To link this version, run:
  brew link go
% brew link go
Linking /usr/local/Cellar/go/1.15.8... 3 symlinks created.
% brew unlink go
Unlinking /usr/local/Cellar/go/1.15.8... 3 symlinks removed.
% brew link go@HEAD-2f0da6d
Error: No such keg: /usr/local/Cellar/go@head-2f0da6d
% brew link go/HEAD-2f0da6d
Error: No available formula with the name "go/head-2f0da6d".
```
Try to search for a solution to this but none worked for me. Can anyone here help?
Thanks!

**Comment by SMillerDev** [comment]
@Homebrew/brew is there a way to `brew link go@HEAD` like that?

**Reply by MikeMcQuaid** [reply]
`brew unlink go; brew link --HEAD go` might do it, I haven't tried (perhaps even without the `brew unlink`).

**Comment by mistydemeo** [comment]
We used to have an official command for this which we discontinued awhile back. I've migrated it into a personal tap, and it may be useful for you. You can find it here: https://github.com/mistydemeo/homebrew-cmd And install it via `brew tap mistydemeo/cmd`

You can see the name of the installed version via `brew info go`, then do `brew switch go HEAD-blah`, where `HEAD-blah` is the specific name of the version Homebrew picked.
